packages/expo-calendar/src/next/Calendar.ts

```diff
@@ -0,0 +1,279 @@
+ * Represents a calendar reminder object that can be accessed and modified using the Expo Calendar Next API.
+ *
+ * @platform ios
+ */
+export class ExpoCalendarReminder extends InternalExpoCalendar.ExpoCalendarReminder {
+  override update(details: Partial<ModifableReminderProperties>): void {
+    const nullableDetailsFields = getNullableDetailsFields(details);
+    super.update(stringifyDateValues(details), nullableDetailsFields);
+  }
+}
+
+/**
+ * Represents a calendar object that can be accessed and modified using the Expo Calendar Next API.
+ *
+ * This class provides properties and methods for interacting with a specific calendar on the device,
+ * such as retrieving its events, updating its details, and accessing its metadata.
+ */
+export class ExpoCalendar extends InternalExpoCalendar.ExpoCalendar {
+  override createEvent(
+    details: Partial<
+      Omit<
+        Event,
+        | 'creationDate'
+        | 'lastModifiedDate'
+        | 'originalStartDate'
+        | 'isDetached'
+        | 'status'
+        | 'organizer'
+      >
+    >
+  ): ExpoCalendarEvent {
+    const newEvent = super.createEvent(stringifyDateValues(details));
+    Object.setPrototypeOf(newEvent, ExpoCalendarEvent.prototype);
+    return newEvent;
+  }
+
+  override createReminder(details: Partial<Reminder>): ExpoCalendarReminder {
+    const newReminder = super.createReminder(stringifyDateValues(details));
+    Object.setPrototypeOf(newReminder, ExpoCalendarReminder.prototype);
+    return newReminder;
+  }
+
+  override listEvents(startDate: Date, endDate: Date): ExpoCalendarEvent[] {
+    if (!startDate) {
+      throw new Error('listEvents must be called with a startDate (date) to search for events');
+    }
+    if (!endDate) {
+      throw new Error('listEvents must be called with an endDate (date) to search for events');
+    }
+    return super.listEvents(stringifyIfDate(startDate), stringifyIfDate(endDate)).map((event) => {
+      Object.setPrototypeOf(event, ExpoCalendarEvent.prototype);
+      return event;
+    });
+  }
+
+  override async listReminders(
+    startDate: Date | null = null,
+    endDate: Date | null = null,
+    status: ReminderStatus | null = null
+  ): Promise<ExpoCalendarReminder[]> {
+    const reminders = await super.listReminders(
+      startDate ? stringifyIfDate(startDate) : null,
+      endDate ? stringifyIfDate(endDate) : null,
+      status
+    );
+    return reminders.map((reminder) => {
+      Object.setPrototypeOf(reminder, ExpoCalendarReminder.prototype);
+      return reminder;
+    });
+  }
+
+  override update(details: Partial<ModifableCalendarProperties>): void {
+    const color = details.color ? processColor(details.color) : undefined;
+    const newDetails = { ...details, color: color || undefined };
+    super.update(newDetails);
+  }
+}
+
+/**
+ * Gets an instance of the default calendar object.
+ * @return An [`ExpoCalendar`](#expocalendar) object that is the user's default calendar.
+ * @platform ios
```

Comment:
If this package is marked iOS only, then all methods/components/types are only available for iOS. I don't think having platform tags individually should be included here or anywhere else in this doc.